### ISSUE TITLE
CognitoConfirmSignOptions metadata not registered 

### ISSUE THREAD

**Issue by Bobby9292** [aws-amplify/amplify-flutter#560]
I have a cognito user pool that is configured to send an email to the user upon creation with a random password, the user needs to create a new password upon first login and also input some optional user attributes ('name 'and 'family_name'). 
So I use Amplify.Auth.confirmSignIn method with the new password as the confirmationValue and I pass in the userAttributes using the CognitoConfirmSignInOptions class like so : 
` Map<String, String> userAttributes = {
            'name': firstName,  
            'family_name': lastName,  
          }; `
          `var options = CognitoConfirmSignInOptions(clientMetadata: userAttributes);  `
      `Amplify.Auth.confirmSignIn(   
          confirmationValue: newPassword, options: options);  `

When I checked the user on Cognito, the attributes were not saved even though the user had successfully passed the confirmation steps. 

**Expected behavior**
- User attributes saved into the Cognito pool along with the user

**Amplify versions:**
- amplify_auth_cognito: ^0.1.4
- amplify_flutter: ^0.1.4

Please let me know if anyone see something I did wrong or if there is any bug at the moment on the package. 
Thanks a lot.

**Comment by haverchuck** [comment]
@Bobby9292 It sounds like you are describing Cognito Lambda Trigger functionality.  Have you written code in the lambda trigger to update the user record with these attributes (I'm not 100% sure if that's actually possible)?  If so, you might want to try updating the lambda trigger code with a log to see if you can observe the data you're passing in getting written to cloudwatch.  Is there any reason why the user cannot pass these attributes in when using the signUp API, when user attributes can be set without the use of a lambda trigger?

**Comment by Bobby9292** [comment]
Hey @haverchuck  

Thanks for you reply and no I don't think it comes from Cognito Lambda Trigger functionality, I left that part to default and when I checked on Cognito all the triggers were set to "none". 

So yes, I didn't explain very well the steps I use to create a user. For now we are in development phase so I create the accounts manually on the cognito web console. Ultimately the policies will be that only admin would be able to create new accounts, a user will NOT be able to sign themselves up thus input all the required user attributes during sign up. 

So when I create a new account I only give the email. The user then receives an email with a password to log into our mobile application or web page. At this point the account status is : _FORCE_CHANGE_PASSWORD_ and since the user did not sign up himself I don't have his name and last name (which are required attributes to complete the sign up process) so on the application side the user needs to add those information along with the new password after logging in for the first time.

I already implemented this functionality for the web app using the Javascript methods and it worked fine I was passing userAttributes as a map directly to the **option** attribute of the Amplify.Auth.completeNewPassword() method.
But here on flutter I need to pass the map to a  ConfirmSignInOptions object first and then pass it to the confirmSignIn() method. Like you mentioned here [https://github.com/aws-amplify/amplify-flutter/issues/532#issuecomment-827039519](url)

Strangely, only the user attributes passed in the options are not updated. The rest works fine. 
Hope this helps you understand a bit more my issue.
